nemo/collections/asr/parts/mixins/interctc_mixin.py

```diff
@@ -0,0 +1,140 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Dict, Tuple
+
+import torch
+
+from nemo.core.classes.mixins import AccessMixin
+
+
+class InterCTCMixin:
+    """Adds utilities for computing interCTC loss from https://arxiv.org/abs/2102.03216.
+
+    To use, make sure encoder defines ``capture_output_at_layers (list[int])``
+    property and registers layer_output_X and layer_length_X for all layers that
+    we want to get loss from. Then call
+
+        * ``setup_interctc`` in the init method
+        * ``self.add_interctc_losses`` after computing regular loss.
+        * ``self.finalize_interctc_metrics(metrics, outputs, prefix="val_")``
+          in the `multi_validation_epoch_end` method.
+        * ``self.finalize_interctc_metrics(metrics, outputs, prefix="test_")``
+          in the `multi_test_epoch_end` method.
+    """
+
```

Comment:
Note: Add method that returns whether interctc mixin is enabled or not rather than assign value to self.